core/vgmgr/virtual_group_manager.go

```diff
@@ -123,3 +123,11 @@ type VirtualGroupManager interface {
 type NewVirtualGroupManager = func(selfOperatorAddress string, chainClient consensus.Consensus) (VirtualGroupManager, error)
 
 type IDSet = map[uint32]struct{}
+
+func NewIDSetFromList(list []uint32) IDSet {
```

Comment:
Generally, core only defines interface, and the implementation can be placed in the base layer.

Reply:
moved